src/cluster_legacy.c

```diff
@@ -4363,11 +4363,15 @@ void clusterSendFailoverAuthIfNeeded(clusterNode *node, clusterMsg *request) {
 
     /* We did not voted for a replica about this primary for two
      * times the node timeout. This is not strictly needed for correctness
-     * of the algorithm but makes the base case more linear. */
-    if (mstime() - node->replicaof->voted_time < server.cluster_node_timeout * 2) {
+     * of the algorithm but makes the base case more linear.
+     *
+     * This limitation does not restrict manual failover. If a user initiates
+     * a manual failover, we need to allow it to vote, otherwise the manual
+     * failover may time out. */
+    if (!force_ack && mstime() - node->replicaof->voted_time < server.cluster_node_timeout * 2) {
         serverLog(LL_WARNING,
-                  "Failover auth denied to %.40s %s: "
-                  "can't vote about this primary before %lld milliseconds",
+                  "Failover auth denied to %.40s (%s): "
+                  "can't vote about this replica before %lld milliseconds",
```

Comment:
I think the text "can't vote about this primary" means to select a new primary for the shard. It is bad English though.

If we want to talk about the replica, we should say "vote for this primary".

```suggestion
                  "Failover auth denied to %.40s (%s): "
                  "can't vote for this replica before %lld milliseconds",
```

But the timestamp `node->replicaof->voted_time` is actually for all replicas of the same primary, so we can't vote for any of them within this time.

I don't know if there is a simle way to write this. It is actually about the primary, so shall we mention the primary node name in the log? Something like this?

```suggestion
                  "Failover auth denied to %.40s (%s): "
                  "can't vote for any replica of %.40s (%s) within %lld milliseconds",
```

Reply:
no, the code in here, can to vote, means we cannot vote for this sender, the sender is a replica. This myself (other primary got the AUTH_REQUEST) has voted for sender (replica) last time, so within the two timeout time, myself (other primary) can not vote for this sender (replica) again.